File: Leetcode/Daily/11-06-2024/prob.go
func relativeSortArray(arr1 []int, arr2 []int) []int {
    mp := make(map[int]int)
    for _,i:=range arr1{
        mp[i]++
    }

    var ans []int
    for _,c := range arr2{
        if count, exists := mp[c]; exists {
			for count > 0 {
				ans = append(ans, c)
				count--
			}
			mp[c] = 0 
		}
    }

    var rems []int
	for num, count := range mp {
		for count > 0 {
			rems = append(rems, num)
			count--
		}
	}

	sort.Ints(rems)
	ans = append(ans, rems...)
    return ans
}

File: Leetcode/Daily/11-06-2024/prob.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        map<int,int>mp;
        for(auto &c:arr1)mp[c]++;

        vector<int>ans;
        for(auto &c:arr2){
            int val = mp[c];
            mp[c] = 0;
            while(val--)ans.push_back(c);
        }
        for(auto &c:mp){
            if(c.second!=0){
                int val = c.second;
                while(val--)ans.push_back(c.first);
            }
        }
        return ans;

        return ans;
    }
};